community/spatial-index/src/main/java/org/neo4j/gis/spatial/index/curves/SpaceFillingCurve.java

```diff
@@ -143,6 +143,23 @@ private Long derivedValueFor( double[] coord, int level )
     {
         assertValidLevel( level );
         long[] normalizedValues = getNormalizedCoord( coord );
+        return derivedValueFor( normalizedValues, level );
+    }
```

Comment:
Worth thinking about. But the current code should be multi-thread safe and the suggested change would break that, which is a disadvantage also worth thinking about.